bank/persistence/user_operations_test.go

```diff
@@ -75,6 +73,31 @@ func TestItChecksSessionIsValidAtAGivenTimeInstant(t *testing.T) {
 	if ok != false {
 		t.Error("expected session to be invalid, but it wasn't")
 	}
+}
+
+func TestItWritesCorrectUserData(t *testing.T) {
```

Comment:
It's smelly mgo's api to be used in the test as it duplicates test and production code. 

Name of the test could be improved to something like: TestAuthenticateExistingUser 

whether the test will do the registration and authentication.